src/lib.rs

```diff
@@ -640,43 +641,106 @@ impl<T: Into<Key>> KeyRange for Range<T> {
     }
 }
 
+#[test]
+fn test_range() -> Result<()> {
+    let input = vec![0]..vec![10];
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), Some(Key::from(vec![9])));
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl<T: Into<Key>> KeyRange for RangeFrom<T> {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         (Bound::Included(self.start.into()), Bound::Unbounded)
     }
 }
 
+#[test]
+fn test_range_from() -> Result<()> {
+    let input = vec![0]..;
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), None);
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl KeyRange for RangeFull {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         (Bound::Unbounded, Bound::Unbounded)
     }
 }
 
+#[test]
+fn test_range_full() -> Result<()> {
+    let input = ..;
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), None);
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl<T: Into<Key>> KeyRange for RangeInclusive<T> {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         let (start, end) = self.into_inner();
         (Bound::Included(start.into()), Bound::Included(end.into()))
     }
 }
 
+#[test]
+fn test_range_inclusive() -> Result<()> {
+    let input = vec![0]..=vec![10];
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), Some(Key::from(vec![10])));
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl<T: Into<Key>> KeyRange for RangeTo<T> {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         (Bound::Unbounded, Bound::Excluded(self.end.into()))
     }
 }
 
+#[test]
+fn test_range_to() -> Result<()> {
+    let input = ..vec![10];
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), Some(Key::from(vec![9])));
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl<T: Into<Key>> KeyRange for RangeToInclusive<T> {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         (Bound::Unbounded, Bound::Included(self.end.into()))
     }
 }
 
+#[test]
+fn test_range_to_inclusive() -> Result<()> {
+    let input = ..=vec![10];
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), Some(Key::from(vec![10])));
+    assert_eq!(output, expected);
+    Ok(())
+}
+
 impl<T: Into<Key>> KeyRange for (Bound<T>, Bound<T>) {
     fn into_bounds(self) -> (Bound<Key>, Bound<Key>) {
         (convert_to_bound_key(self.0), convert_to_bound_key(self.1))
     }
 }
 
+#[test]
+fn test_bound_bound() -> Result<()> {
+    let input = (Bound::Included(vec![0]), Bound::Included(vec![10]));
+    let output = input.into_keys()?;
+    let expected = (Key::from(vec![0]), Some(Key::from(vec![10])));
+    assert_eq!(output, expected);
+    Ok(())
+}
+
```

Comment:
Could you move all the tests to the test mod please?

Reply:
Yes!